webapp/components/signup/components/signup_ldap.jsx

```diff
@@ -0,0 +1,404 @@
+// Copyright (c) 2015 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import FormError from 'components/form_error.jsx';
+
+import * as GlobalActions from 'actions/global_actions.jsx';
+import {track} from 'actions/analytics_actions.jsx';
+
+import BrowserStore from 'stores/browser_store.jsx';
+import UserStore from 'stores/user_store.jsx';
+
+import * as Utils from 'utils/utils.jsx';
+import Client from 'client/web_client.jsx';
+
+import React from 'react';
+import {FormattedMessage, FormattedHTMLMessage} from 'react-intl';
+import {browserHistory, Link} from 'react-router/es6';
+
+import logoImage from 'images/logo.png';
+
+export default class SignupLdap extends React.Component {
+    static get propTypes() {
+        return {
+            location: React.PropTypes.object
+        };
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.handleLdapSignup = this.handleLdapSignup.bind(this);
+
+        this.handleLdapIdChange = this.handleLdapIdChange.bind(this);
+        this.handleLdapPasswordChange = this.handleLdapPasswordChange.bind(this);
+
+        this.state = {
+            data: '',
+            hash: '',
+            usedBefore: false,
+            email: '',
+            teamDisplayName: '',
+            teamName: '',
+            teamId: '',
+            openServer: false,
+            loading: true,
+            inviteId: '',
+            ldapId: '',
+            ldapPassword: ''
+        };
+    }
+
+    componentWillMount() {
+        let data = this.props.location.query.d;
+        let hash = this.props.location.query.h;
+        const inviteId = this.props.location.query.id;
+        let usedBefore = false;
+        let email = '';
+        let teamDisplayName = '';
+        let teamName = '';
+        let teamId = '';
+        let openServer = false;
+        let loading = true;
+
+        if ((inviteId && inviteId.length > 0) || (hash && hash.length > 0)) {
```

Comment:
You only need to check once if the user is already logged in in the SignupController instead of checking in both SignupEmail and SignupLdap. That way, it'll save the user a click when they're already logged in, and we also won't be duplicating that code.